modules/geolocationRtdProvider.js

```diff
@@ -0,0 +1,64 @@
+import {submodule} from '../src/hook.js';
+import {isFn, logError, pick, cleanObj} from '../src/utils.js';
+import {config} from '../src/config.js';
+import {getGlobal} from '../src/prebidGlobal.js';
+import '../src/adapterManager.js';
+
+let permissionsAvailable = true;
+let geolocation;
+function getGeolocationData(requestBidsObject, onDone, providerConfig, userConsent) {
+  let done = false;
+  if (!permissionsAvailable) return complete();
+  const requestPermission = providerConfig.requestPermission === true;
+  const waitForIt = providerConfig.waitForIt;
+  const auctionDelay = config.getConfig('realTimeData.auctionDelay') || 0;
+  const adUnits = requestBidsObject.adUnits || getGlobal().adUnits || [];
+  navigator.permissions.query({
+    name: 'geolocation',
+  }).then(permission => {
+    if (permission.state !== 'granted' && !requestPermission) return complete();
+    navigator.geolocation.getCurrentPosition(geo => {
+      geolocation = pick(geo.coords, [
+        'accuracy',
+        'altitude',
+        'altitudeAccuracy',
+        'heading',
+        'latitude',
+        'longitude',
+        'speed'
+      ]);
+      complete();
+    });
+  });
+  if (auctionDelay > 0) {
+    setTimeout(complete, auctionDelay);
```

Comment:
the rtd parent module handles auction delay and waitforit, no need for you to

Reply:
see https://github.com/AdmixerTech/Prebid.js/blob/c57da4b7f2c7d4cb543e076e34aec70018443bae/modules/rtdModule/index.js#LL306C1-L308C1